tests/testImageDecorrelation.py

```diff
@@ -0,0 +1,253 @@
+    @param verbose be verbose
+
+    @return im1, im2, im1_psf, im2_psf: the template and science images (as 2-d numpy.arrays) and
+    their corresponding PSFs (also as 2-d numpy.arrays)
+
+    @note having sources near the edges really messes up the
+    fitting (probably because of the convolution). So make sure no
+    sources are near the edge.
+    @note also it seems that having the variable source with a large
+    flux increase also messes up the fitting (seems to lead to
+    overfitting -- perhaps to the source itself). This might be fixed by
+    adding more constant sources.
+    """
+    np.random.seed(seed)
+
+    psf1 = [2.2, 2.2] if psf1 is None else psf1
+    if not hasattr(psf1, "__len__"):
+        psf1 = [psf1, psf1]
+    psf2 = [3.3, 3.3] if psf2 is None else psf2
+    if not hasattr(psf2, "__len__"):
+        psf2 = [psf2, psf2]
+    offset = [0., 0.] if offset is None else offset   # astrometric offset (pixels) between the two images
+    if verbose:
+        print('Template PSF:', psf1, theta1)
+        print('Science PSF:', psf2, theta2)
+        print(np.sqrt(psf2[0]**2 - psf1[0]**2))
+        print('Offset:', offset)
+
+    xim = np.arange(-256, 256, 1) if xim is None else xim
+    yim = xim.copy() if yim is None else yim
+    x0im, y0im = np.meshgrid(xim, yim)
+    fluxes = np.random.uniform(50, 30000, n_sources)
+    xposns = np.random.uniform(xim.min()+16, xim.max()-5, n_sources)
+    yposns = np.random.uniform(yim.min()+16, yim.max()-5, n_sources)
+
+    # Make the source closest to the center of the image the one that increases in flux
+    ind = np.argmin(xposns**2. + yposns**2.)
+
+    im1 = np.random.normal(scale=sig1, size=x0im.shape)  # sigma of template
+    im2 = np.random.normal(scale=sig2, size=x0im.shape)  # sigma of science image
+
+    # variation in y-width of psf in science image across (x-dim of) image:
+    psf2_yvary = psf_yvary_factor * (yim.mean() - yposns) / yim.max()
+    if verbose:
+        print('PSF y spatial-variation:', psf2_yvary.min(), psf2_yvary.max())
+
+    for i in range(n_sources):
+        flux = fluxes[i]
+        tmp1 = flux * singleGaussian2d(x0im, y0im, xposns[i], yposns[i], psf1[0], psf1[1], theta=theta1)
+        im1 += tmp1
+        if i == ind:
+            flux += flux * varSourceChange
+        tmp2 = flux * singleGaussian2d(x0im, y0im, xposns[i]+offset[0], yposns[i]+offset[1],
+                                       psf2[0], psf2[1]+psf2_yvary[i], theta=theta2)
+        im2 += tmp2
+
+    # Add a (constant, for now) background offset to im2
+    if im2background != 0.:  # im2background = 10.
+        if verbose:
+            print('Background:', im2background)
+        im2 += im2background
+
+    im1_psf = singleGaussian2d(x0im, y0im, 0, 0, psf1[0], psf1[1], theta=theta1)
+    im2_psf = singleGaussian2d(x0im, y0im, offset[0], offset[1], psf2[0], psf2[1], theta=theta2)
+    return im1, im2, im1_psf, im2_psf
```

Comment:
Would it make sense to return exposures instead of images? Then you would have a place to put the PSFs and you'd have mask planes and variance planes, should your test ever want to deal with them.